gk/main.c

```diff
@@ -174,14 +174,14 @@ look_up_fib(struct gk_lpm *ltbl, struct ip_flow *flow)
 	int fib_id;
 
 	if (flow->proto == ETHER_TYPE_IPv4) {
-		fib_id = lpm_lookup_ipv4(ltbl->lpm, flow->f.v4.dst);
+		fib_id = lpm_lookup_ipv4(ltbl->lpm, flow->f.v4.dst.s_addr);
 		if (fib_id < 0)
 			return NULL;
 		return &ltbl->fib_tbl[fib_id];
 	}
 
 	if (likely(flow->proto == ETHER_TYPE_IPv6)) {
-		fib_id = lpm_lookup_ipv6(ltbl->lpm6, flow->f.v6.dst);
+		fib_id = lpm_lookup_ipv6(ltbl->lpm6, flow->f.v6.dst.s6_addr);
```

Comment:
Update the type of the second parameter of `lpm_lookup_ipv6()` from `uint8_t *ip` to `struct in6_addr *ip`. Calling `rte_lpm6_lookup()` in `lpm_lookup_ipv6()` will require a cast, but the calls of `lpm_lookup_ipv6()` will be safe.